### PR TITLE
new macros

### PR DESCRIPTION
**Pull request by fperrad**
this serie introduces 2 new macros

### CODE REVIEW

File: tommath.h [hunk 1]
```diff
@@ -147,8 +147,10 @@ TOOM_SQR_CUTOFF;
 #   endif
 #endif
 
+#define MP_TYPE_BIT(type) (CHAR_BIT * sizeof(type))
```
Comment (minad) [inline tommath.h]: MP_TYPE_BIT should be defined in tommath_private.h

File: tommath.h [hunk 2]
```diff
@@ -147,8 +147,10 @@ TOOM_SQR_CUTOFF;
...
+
 /* size of comba arrays, should be at least 2 * 2**(BITS_PER_WORD - BITS_PER_DIGIT*2) */
-#define PRIVATE_MP_WARRAY (1u << (((CHAR_BIT * sizeof(mp_word)) - (2 * MP_DIGIT_BIT)) + 1))
+#define PRIVATE_MP_WARRAY (1u << ((MP_TYPE_BIT(mp_word) - (2 * MP_DIGIT_BIT)) + 1))
```
Comment (minad) [inline tommath.h]: Just don't use MP_TYPE_BIT here since PRIVATE_MP_WARRAY will be moved to tommath_private.h later.

File: tommath.h (hunk 1 above)
Comment (minad) [reply tommath.h]: I would also suggest MP_SIZEOF_BITS instead

File: bn_s_mp_exptmod_fast.c [hunk 3]
```diff
@@ -85,7 +85,7 @@ int s_mp_exptmod_fast(const mp_int *G, const mp_int *X, const mp_int *P, mp_int
       /* automatically pick the comba one if available (saves quite a few calls/ifs) */
 #ifdef BN_S_MP_MONTGOMERY_REDUCE_FAST_C
       if ((((P->used * 2) + 1) < (int)MP_WARRAY) &&
-          (P->used < (1 << ((CHAR_BIT * sizeof(mp_word)) - (2 * MP_DIGIT_BIT))))) {
+          (P->used < (1 << MP_MAXFAST_BIT))) {
```
Comment (minad) [inline bn_s_mp_exptmod_fast.c]: Would it make sense to introduce another macro MP_MAXFAST for (1 << MP_MAXFAST_BIT)?

**Comment by fperrad** [comment]
rebased + improved

File: bn_mp_prime_is_prime.c [hunk 4]
```diff
@@ -287,11 +287,11 @@ int mp_prime_is_prime(const mp_int *a, int t, int *result)
           * One 8-bit digit is too small, so concatenate two if the size of
           * unsigned int allows for it.
           */
-         if (((sizeof(unsigned int) * CHAR_BIT)/2) >= (sizeof(mp_digit) * CHAR_BIT)) {
+         if ((MP_TYPE_BIT(unsigned int)/2) >= MP_SIZEOF_BITS(mp_digit)) {
```
Comment (minad) [inline bn_mp_prime_is_prime.c]: MP_TYPE_BIT?

Comment (fperrad) [reply bn_mp_prime_is_prime.c]: @minad fixed

**Comment by minad** [comment]
@sjaeckel I think #238, #241, #242, #244 and #249 are ready for merge.

**Comment by sjaeckel** [comment]
> @sjaeckel I think #238, #241, #242, #244 and #249 are ready for merge.

#242 and #249 are merged, the rest has conflicts, can you all please rebase

**Comment by fperrad** [comment]
rebased

**Pull request merged**